Adding a contributing guide (#11)
jenkinsci/saferestart-plugin: CONTRIBUTING.md

# Built-on-column-plugin

This plugin Adds a column on the jenkins dashboard that shows the actual node the last build was run on.

# Why should you contribute

> You can contribute in many ways, and whatever you choose we're grateful! Source code contribution is the obvious one but we also need your feedback and if you don't really want to participate in the implementation directly you may still have great ideas about features we need (or should get rid of).

Reviewer: I prefer sentence per line because it makes code review easier.  Asking for suggestions of feature removals is much less likely to have positive responses, since feature removal is a breaking change and we avoid breaking changes.
```suggestion
You can contribute in many ways, and whatever you choose we're grateful
Source code contribution is the obvious one but we also need your feedback and if you don't really want to participate in the implementation directly you may still have great ideas about features we need.
```

> We have our vision for the plugin and we have an experience with maintaining Jenkins instances, but the plugin is not supposed to solve only our problems. Surely we haven't experienced all of them... That's why we want to hear from you.

Reviewer: Sentence per line
```suggestion
We have our vision for the plugin and we have an experience with maintaining Jenkins instances, but the plugin is not supposed to solve only our problems.
Surely we haven't experienced all of them...
That's why we want to hear from you.
```

> Please use GitHub issues if you need to report a bug or request changes/improvements. Whenever you report a problem please provide information about:

Reviewer: This plugin uses Jira for issue tracking, not GitHub.
```suggestion
Please use Jenkins Jira if you need to [report a bug](https://www.jenkins.io/participate/report-issue/redirect/#16072) or [request changes/improvements](https://issues.jenkins.io/secure/CreateIssueDetails!init.jspa?pid=10172&issuetype=2&priority=4&components=16072). Whenever you report a problem please provide information about:
```

> * Plugin version (Configuration as Code plugin as well any other plugin you suspect your problem to be related to)
> * Jenkins version
> * Operating system

Reviewer: More details on the technique to report Jenkins issues should be linked here
```suggestion
* Jenkins version, operating system version, and plugin versions (using the script from ["How to report an issue"](https://www.jenkins.io/participate/report-issue/#Howtoreportanissue-WhatinformationtoprovideforEnvironmentandDescription))
```

* Description!

# Run Locally

Prerequisites: Java, Maven 

>  * Ensure Java 8 or 11 is available.

Reviewer: This plugin no longer supports Java 8
```suggestion
 * Ensure Java 11 or 17 is available.
```

```bash
  $ java -version	
```

- Ensure Maven is included in the PATH environment variable.

```bash
  export PATH=$PATH:/path/to/apache-maven-3.8.6/bin
```
* To run a plugin locally, the command is:

```bash
  mvn hpi:run
```

> * That starts a Jenkins controller with the minimum Jenkins version supported by the plugin and loads the plugin and its dependencies into that controller. Jenkins will be running on port 8080 and can be reached with the URL http://localhost:8080/jenkins/

Reviewer: Sentence per line
```suggestion
* That starts a Jenkins controller with the minimum Jenkins version supported by the plugin and loads the plugin and its dependencies into that controller.
  Jenkins will be running on port 8080 and can be reached with the URL http://localhost:8080/jenkins/
```

# Built-on-column-plugin

> This plugin Adds a column on the jenkins dashboard that shows the actual node the last build was run on.

Reviewer: Capitalize Jenkins (it is a proper noun)
```suggestion
This plugin Adds a column on the Jenkins dashboard that shows the actual node the last build was run on.
```

# Why should you contribute

You can contribute in many ways, and whatever you choose we're grateful
Source code contribution is the obvious one but we also need your feedback and if you don't really want to participate in the implementation directly you may still have great ideas about features we need.

We have our vision for the plugin and we have an experience with maintaining Jenkins instances, but the plugin is not supposed to solve only our problems.
Surely we haven't experienced all of them...
That's why we want to hear from you.

> Please use Jenkins Jira if you need to [report a bug](https://www.jenkins.io/participate/report-issue/redirect/#16072) or [request changes/improvements](https://issues.jenkins.io/secure/CreateIssueDetails!init.jspa?pid=10172&issuetype=2&priority=4&components=16072). Whenever you report a problem please provide information about:

Reviewer: Sentence per line that I missed earlier
```suggestion
Please use Jenkins Jira if you need to [report a bug](https://www.jenkins.io/participate/report-issue/redirect/#16072) or [request changes/improvements](https://issues.jenkins.io/secure/CreateIssueDetails!init.jspa?pid=10172&issuetype=2&priority=4&components=16072).
Whenever you report a problem please provide information about:
```
